File: API/Models/Dish.cs
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;

namespace API.Models
{
    public class Dish
    {
        public int Id { get; set; }
        public string Name { get; set; } = string.Empty;
        public string Description { get; set; } = string.Empty;
        public int Quantity { get; set; }
        [Column(TypeName = "decimal(18,2)")]
        public decimal Price { get; set; }
        public string ImageUrl { get; set; } = string.Empty;
        public bool IsAvailable { get; set; } = true;

        public int RestaurantId { get; set; } // Foreign key to Restaurant
        public required virtual Restaurant Restaurant { get; set; } // Navigation property to Restaurant

        public int? CategoryId { get; set; } // Optional foreign key to Category
        public virtual Category? Category { get; set; } // Navigation property to Category

        public ICollection<OrderDish> OrderItems { get; set; } = new List<OrderDish>(); // Navigation property for join table
    }
}

File: careem_clone_mobile/lib/models/dish.dart
class Dish {
  final int id;
  final String name;
  final String? description;
  final double price;
  final String? image;
  final int categoryId;
  
  Dish({
    required this.id,
    required this.name,
    this.description,
    required this.price,
    this.image,
    required this.categoryId,
  });
  
  factory Dish.fromJson(Map<String, dynamic> json) {
    return Dish(
      id: json['id'],
      name: json['name'],
      description: json['description'],
      price: json['price']?.toDouble() ?? 0.0,
      image: json['imageUrl'],
      categoryId: json['categoryId'],
    );
  }
}
